server/commands/expand.go

```diff
@@ -43,21 +40,44 @@ func (query *ExpandQuery) Execute(ctx context.Context, req *openfgapb.ExpandRequ
 
 	tk := tupleUtils.NewTupleKey(object, relation, "")
 
-	userset, err := query.getUserset(ctx, store, modelID, tk)
+	model, err := query.datastore.ReadAuthorizationModel(ctx, store, modelID)
 	if err != nil {
-		return nil, err
+		if errors.Is(err, storage.ErrNotFound) {
+			return nil, serverErrors.AuthorizationModelNotFound(modelID)
+		}
+
+		return nil, serverErrors.HandleError("", err)
 	}
 
-	model, err := query.datastore.ReadAuthorizationModel(ctx, req.GetStoreId(), req.GetAuthorizationModelId())
+	typesys := typesystem.New(model)
+
+	err = validation.ValidateObject(typesys, tk)
 	if err != nil {
-		if err == storage.ErrNotFound {
-			return nil, serverErrors.AuthorizationModelNotFound(req.GetAuthorizationModelId())
+		return nil, serverErrors.HandleTupleValidateError(err)
+	}
+
+	err = validation.ValidateRelation(typesys, tk)
+	if err != nil {
+		return nil, serverErrors.HandleTupleValidateError(err)
+	}
+
+	objectType := tupleUtils.GetType(object)
+	rel, err := typesys.GetRelation(objectType, relation)
+	if err != nil {
+		if errors.Is(err, typesystem.ErrObjectTypeUndefined) {
+			return nil, serverErrors.TypeNotFound(objectType)
+		}
+
+		if errors.Is(err, typesystem.ErrRelationUndefined) {
+			return nil, serverErrors.RelationNotFound(relation, objectType, tk)
 		}
```

Comment:
This isn't covered by tests, but we can remove it because `validation.ValidateRelation(typesys, tk)` already throws these errors

Reply:
We have to get the `*openfgapb.Relation` from the typesystem by calling `typesys.GetRelation` because we use it for the rewrite rule.

```
rel, err := typesys.GetRelation(...)
if err != nil {
  // this error handling code
}

userset := rel.GetRewrite()

// use the 'userset' for Expand
```

So though this error handling code is redundant because presumably you know the type and relation exists (otherwise the validation code above would have thrown an error), it's still the appropriate error handling mechanism here. If we hit an error in this code path that we didn't hit above in our validation code path then that's a serious issue. 

I'm happy to change this to a panic instead, but otherwise we should leave as is with the appropriate error handling as it is now.

Reply:
Future PR!